### ISSUE TITLE
Please implement Inventory Tweaks sorting in the grid

### ISSUE THREAD

**Issue by ancilla-x** [raoulvdberge/refinedstorage#899]
Title, I know that AE2 for 1.7.10 used to have this option, and I'd love to see it brought here. Currently on 1.10.2, RS version 1.2.14.

**Comment by raoulvdberge** [comment]
Incomplete. Define what "inventory tweaks sorting" is - exactly, please.

**Comment by ancilla-x** [comment]
Sorry, not sure how else to describe it. The mod "Inventory Tweaks" has a specific way of sorting items in an inventory. 

https://inventory-tweaks.readthedocs.io/en/latest/
https://github.com/Inventory-Tweaks/inventory-tweaks/wiki

I would like to see this way of sorting items as an option in the Grid for RS, alongside "A-Z" and "Amount".

**Comment by raoulvdberge** [comment]
How does Inventory Tweaks sort those items?

**Comment by ancilla-x** [comment]
I don't really understand the question, I'm not a developer so I don't have any knowledge of sorting algorithms etc, but I linked to the github of the mod which should contain all the documentation on how it works. 

I asked the original question because if anyone uses this mod, and it's present in most big release modded packs, their inventories and chests are sorted via this mod and display the items in a certain order. I would like to see that order present in RS so I can choose to show my items in the order I have in my chests/inventory.

**Comment by raoulvdberge** [comment]
Let me rephrase the question:

A-Z is sorted alphabetically, "amount" is sorted by item count, so how is Inventory Tweaks being sorted?

**Comment by way2muchnoise** [comment]
It is some silly config file, no clue if the actual sorting method is exposed

**Comment by ancilla-x** [comment]
[InvTweaksTree.txt](https://github.com/raoulvdberge/refinedstorage/files/722260/InvTweaksTree.txt)

So I've checked the mod config folder and found this, looks like the sorting rules. Hopefully it's what you're asking after, 

@way2muchnoise, there's no need to call something silly just because you don't agree with how it's done.

**Comment by Joshwoo70** [comment]
yeah true. i actually don't mind having jnvtweaks sorting same as the original AE2.

**Comment by raoulvdberge** [comment]
I don't understand that XML file, so if anyone is able to explain how InvTweaks *exactly* sorts, enlighten me.

**Comment by way2muchnoise** [comment]
I guess this would do? https://github.com/Inventory-Tweaks/inventory-tweaks/blob/develop/src/main/java/invtweaks/api/InvTweaksAPI.java#L70

**Comment by raoulvdberge** [comment]
aha, thanks! I tried to set up a dev instance with InvTweaks but it kept crashing, but that should make it real easy to do.

**Comment by choekstr** [comment]
Ahh, the real reason mousewheel support couldn't be added...the dev environment kept crashing... Kappa

**Comment by ancilla-x** [comment]
Awesome guys, let it never be said you don't listen to your userbase. If this can be implemented easily enough then great stuff.

**Comment by raoulvdberge** [comment]
I will look into this week, but I first gotta try to launch MC with InvTweaks without crashing :D

**Comment by choekstr** [comment]
You two are indeed the best but then again you know that as I say it all the time!  :)
And really surprised you get a crash with it as I add it to every single pack since 1.7 days and haven't seen an issue.  Guess I have been really lucky.  

I *think* the Cntrl-Shift click to move all like items/stacks into a chest is a killer feature that I think comes from InvTweaks and would be great to add as well.  I keep trying that everytime I open a grid but we do have the MC default of picking up the item and shift double clicking to do a similar function.

**Comment by raoulvdberge** [comment]
@choekstr I know you'd kill for that feature (and the mouse scrolling stuff) but I simply think that the InvTweaks API doesn't allow it. Then there is not much I can do.
